### ISSUE TITLE
Bug: sam deploy no longer zipping dependencies

### ISSUE THREAD

**Issue by mikegrima** [aws/aws-sam-cli#5008]
### Description:
Hello:

I'm pulling my hair out trying to understand why `sam deploy` is no longer zipping my dependencies. I upgraded to the latest version, and then also tried downgrading to an older version and no change in behavior!

I literally have no idea where to look because I even downgraded to an older version of SAM CLI `1.74.0`, and it's doing the exact same thing. My lambda packages used to be over 12MB and now they are only 190KB and failing when running because it can't find any packages. I'm building a Python 3.9 project.

### Steps to reproduce:
First run SAM build. This is working properly and in the `.aws-sam/build` directory, it is correctly placing all the dependencies in there. 

`sam deploy` just uploads my code without any of the dependencies whatsoever.

### Observed result:
The output shows absolutely nothing about zipping files. It shows that it is uploading files when they are not present in S3, and it's showing the upload filesize as being about 190KB.

### Expected result:
My lambda has all the dependencies packaged as part of the Zip.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)
I'm running on macOS and building a Python 3.9 project.

1. OS: macOS
2. `sam --version`: SAM CLI, version 1.74.0
3. AWS region: us-east-2
4. 

```
# Paste the output of `sam --info` here
```
```
{
  "version": "1.74.0",
  "system": {
    "python": "3.9.15",
    "os": "macOS-13.3.1-arm64-arm-64bit"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.24",
    "aws_cdk": "2.73.0 (build 43e681e)",
    "terraform": "1.4.2"
  }
}
```

**Comment by mikegrima** [comment]
Here is the `pip freeze` output because I installed an older SAM CLI in Python:
```
arrow==1.2.3
attrs==22.2.0
aws-lambda-builders==1.27.0
aws-sam-cli==1.74.0
aws-sam-translator==1.59.0
binaryornot==0.4.4
boto3==1.26.112
botocore==1.29.112
certifi==2022.12.7
cffi==1.15.1
cfn-lint==0.74.0
chardet==4.0.0
chevron==0.14.0
click==8.1.3
cookiecutter==2.1.1
cryptography==39.0.2
dateparser==1.1.8
docker==4.2.2
Flask==2.0.3
idna==2.10
itsdangerous==2.1.2
Jinja2==3.1.2
jinja2-time==0.2.0
jmespath==0.10.0
jschema-to-python==1.2.3
jsonpatch==1.32
jsonpickle==3.0.1
jsonpointer==2.3
jsonschema==4.17.3
junit-xml==1.9
MarkupSafe==2.1.2
networkx==3.1
pbr==5.11.1
pycparser==2.21
pydantic==1.10.7
pyOpenSSL==23.0.0
pyrsistent==0.19.3
python-dateutil==2.8.2
python-slugify==8.0.1
pytz==2023.3
PyYAML==5.4.1
regex==2021.9.30
requests==2.25.1
ruamel.yaml==0.17.21
ruamel.yaml.clib==0.2.7
s3transfer==0.6.0
sarif-om==1.0.4
serverlessrepo==0.1.10
six==1.16.0
text-unidecode==1.3
tomlkit==0.7.2
typing_extensions==4.4.0
tzlocal==3.0
urllib3==1.26.15
watchdog==2.1.2
websocket-client==1.5.1
Werkzeug==2.0.3
```

**Comment by mikegrima** [comment]
The way that I'm deploying this right is that I need to build it, then run `sam package` so that it generates what the name of the artifact is going to be in S3. After that, I then need to go to the `.aws-sam/` directory, go into one of my lambda function names, then zip up everything in that directory, rename the .zip with the name of the zipped object in S3, and _then_ run `aws sam deploy`...

**Comment by mikegrima** [comment]
Wow so I figured out thanks to this video: https://www.youtube.com/watch?v=bNbBd6XoDHg

I recently (a month or 2 but haven't done a deployment in a while) made some changes to my `samconfig.toml` file where I have a `TEST` and `PROD` configuration.

For `[TEST.build.parameters]`, I specify:
```
[TEST.build.parameters]
use_container = true
template_file = "template_test.yaml"
```

I copied `template_file = "template_test.yaml"` to `deploy` and `package`, and my guess is that was making it fail to locate the cloned `template.yaml` inside of the `.aws-sam/...` path. Because it couldn't locate the `template.yaml` there, it defaulted to the `CodeUri` path. This is explained in the video above at around the 13:40 mark.

I would like to leave this here in case anyone else pulls their hair out trying to figure this out!
